### ISSUE TITLE
JAWS is announcing "button" as control type for "Modified" columnheader while navigating in virtual pc cursor mode  of the DetailsList

### ISSUE THREAD

**Issue by v-vibsingh** [microsoft/fluentui#16788]
JAWS is announcing "button" as control type for "Modified" columnheader while navigating in virtual pc cursor mode
 of the DetailsList

**Environment Information**
For Jaws: Chrome Version 87.0.4280.88 (Official Build) (64-bit)

**Describe the issue:**
JAWS is announcing "button" as control type for "Modified" columnheader while navigating in virtual pc cursor mode
 of the DetailsList

**Reproduction of the issue in a codepen:**
https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist
issue_detail_list

![image](https://user-images.githubusercontent.com/76475047/106745222-cdb70680-6646-11eb-8d08-6129e5431c8d.png)

**Actual behavior:**
JAWS is announcing "button" as control type for "Modified" columnheader while navigating in virtual pc cursor mode

**Expected behavior:**
JAWS should announce "columnheader" as control type for "Modified" columnheader while navigating in virtual pc cursor mode

**Comment by v-vibsingh** [comment]
Hi @joschect,

Please provide the update on this issue.
This issue is related to SLA based bug on our project. Could you please increase it's priority?

Regards,
Vibhas

**Comment by joschect** [comment]
@v-vibsingh Sorry for the delay! We'll take a look at this.

**Comment by joschect** [comment]
@v-vibsingh After looking into this, I believe that this is intentional  and is working as intended. In this [Grid Example from W3](https://www.w3.org/TR/wai-aria-practices/examples/grid/dataGrids.html) it sounds like `role, button` is read out when focus is put on the sortable date header (see image below). I think this is because button needs to be read out so users know that an element can be interacted with.

I might be missing something, could you provide more detail as to why this is failing accessibility for you?

![image](https://user-images.githubusercontent.com/20801176/109187098-6de3e500-7746-11eb-9f01-e8b87cae6d6b.png)

**Comment by joschect** [comment]
@smhigley I believe that this control having a role button is correct because it can be interacted with, could you confirm?

**Comment by smhigley** [comment]
There's a short answer to this and a long answer :). The short answer is that it is expected behavior, and JAWS is saying the correct thing for a column header that contains a button.

The longer answer has to do with how grids are a bit different from every other composite widget (e.g. tabs, trees, and listboxes). Unlike other composite widgets, grid cells are both interactive themselves, and allowed to contain other interactive items. Almost no other interactive elements or roles allow this, so it comes with some quirks. The principal difficulty has to do with keyboard interaction, but presentation of the gridcell > button is another part of the UX that differs between screen readers.

The other thing at play is that grid is essentially an interactive version of table, and gridcells have no assumed default interaction beyond maybe editing content. As a counter example, when focus is on a tab or listbox option, the user knows that the default interaction (i.e. click or enter/space) will select the current tab/option. With a gridcell, that isn't true, and it isn't always even assumed that a gridcell will have any interaction at all. That's why using a button within the gridcell makes sense for cells that perform a button-style action, e.g. toggling the sorting state.

It would certainly be possible to remove the button and handle all interaction on the element with `role="gridcell"`, but that has two major drawbacks: one: as mentioned above, it isn't obvious that the gridcell can be clicked/activated, and two: even if a screen reader user is aware they can interact, it isn't easy to trigger a click on a gridcell with virtual cursor on. That's why overall, using a button within a gridcell instead of a gridcell alone makes sense :).

Hope that all makes sense! I'm going to close this issue, but feel free to reach out or reopen it if you run into any other issues!

**Comment by v-vibsingh** [comment]
@smhigley

Thanks for the info.

There is discussion going on this issue with the test team and their SME's of the Project.
After their discussion will reopen this issue if needed.

**Comment by smhigley** [comment]
@v-vibsingh feel free to tell them to ping me if the disagree. That'll probably be faster than going back and forth on the issue :)
